File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int result = 0;
        int start = 0;
        int end = arr.length-1;
        while(start<end){
            int mid = start + (end-start)/2;
            if(arr[mid]<arr[mid+1]){
                start = mid+1;
            }
            else{
                end = mid;
            }
        }
        return end;
    }
}

File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var peakIndexInMountainArray = function(arr) {
    let start = 0, end = arr.length-1;
    while(start<end){
        let mid = Math.floor(start+(end-start)/2);
        if(arr[mid] > arr[mid+1]){
            end = mid;
        }else{
            start = mid+1;
        }
    }
    return end;
};